### DISCUSSION TITLE
Suitable detection algorithm for viewpoint change detection

### DISCUSSION THREAD

**Discussion by SAGNIKMJR** [Breakthrough/PySceneDetect#412] [Detection Algorithms]
Hi, does anybody have any idea which detection algorithm might be best suited for detecting viewpiont changes (e.g., closeup vs. distant, first- vs. third-person, etc. )? I want the detector to be triggered only when the viewpoint changes, but not when the content changes in the same viewpoint. Thanks!

**Comment by wjs018** [comment]
So, if I understand this correctly, you are only interested in detecting cuts in which most of the content in the frame is similar in some attributes (like color), but different in others (like edges) and ignore any detected cuts where both of these change. I don't think that by default you would be able to set up detectors to do this.

Instead, what I would recommend is to analyze a video with a number of different detectors and save the analysis to a statsfile. You would then need to run your own analysis on that statsfile to figure out which transitions match the pattern you want and which ones don't and can be ignored.

As an example, let's say you are interested in detecting when the viewpoint switches between two different angles that are both looking down on an athletic field. In that case, the color distribution of the resulting video is likely to be unchanged between the two shots, meaning that it is unlikely to be detected using `detect-hist` or `detect-content` (using the color weights). However, if all the players, stands, and lines are moved around on screen, then the edges within the frame are different, so it might trigger a cut with `detect-content` (using only the edges weight) or `detect-hash`.

So, in that case, you would analyze your video while saving out the stats for those different detectors to a statsfile, then analyze the statsfile afterwards using whatever criteria you want. This bit is outside the scope of this project though.

**Comment by SAGNIKMJR** [comment]
@wjs018, got it. Thanks a lot for the suggestion!
